jetty-io/src/main/java/org/eclipse/jetty/io/ssl/SslConnection.java

```diff
@@ -790,6 +809,9 @@ public int fill(ByteBuffer buffer) throws IOException
                     }
                     catch (Throwable x)
                     {
+                        // Clear the input buffers so that releaseInputBuffers()
+                        // in the finally block will not leak them.
+                        clearInputBuffers();
```

Comment:
I think a better solution would be to javadoc that "releaseInputBuffers" only releases buffers if they are not empty. Then here use a method "discardInputBuffers()" which is javadoc'd to say that it will release the buffer EVEN if not empty.

Or perhaps rename `releaseInputBuffers` to `releaseEmptyInputBuffers` and then have a `releaseInputBuffers` that always releases.    Eitherway, I don't like clearing to influence a later call - just release them here.

Reply:
done.